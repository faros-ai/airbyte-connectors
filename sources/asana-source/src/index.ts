import {Command} from 'commander';
import {
  AirbyteSourceBase,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import {AirbyteSourceLogger} from 'faros-airbyte-cdk';
import VError from 'verror';

import {Asana, AsanaConfig} from './asana';
import {
  Projects,
  ProjectTasks,
  Tags,
  Tasks,
  Users,
  Workspaces,
} from './streams';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteSourceLogger();
  const source = new AsanaSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** Asana source implementation. */
export class AsanaSource extends AirbyteSourceBase<AsanaConfig> {
  get type(): string {
    return 'asana';
  }

  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(config: AsanaConfig): Promise<[boolean, VError]> {
    try {
      const asana = Asana.instance(config, this.logger);
      await asana.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: AsanaConfig): AirbyteStreamBase[] {
    return [
      new Projects(config, this.logger),
      new Tags(config, this.logger),
      new Tasks(config, this.logger),
      new Users(config, this.logger),
      new Workspaces(config, this.logger),
      new ProjectTasks(config, this.logger),
    ];
  }
}
