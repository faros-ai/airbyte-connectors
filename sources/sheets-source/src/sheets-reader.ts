import {GetObjectCommand, S3Client, S3ClientConfig} from '@aws-sdk/client-s3';
import {AirbyteLogger} from 'faros-airbyte-cdk';
import {
  GoogleSpreadsheet,
  GoogleSpreadsheetRow,
  GoogleSpreadsheetWorksheet,
  ServiceAccountCredentials,
} from 'google-spreadsheet';
import {VError} from 'verror';
import XLSX from 'xlsx';

type GoogleCreds = string | ServiceAccountCredentials;

type LocalSpreadsheet = {
  sheet_source: 'LocalSpreadsheet';
  file: string;
};

type GoogleSheet = {
  sheet_source: 'GoogleSheet';
  google_sheet_id: string;
  google_service_account_private_key?: string;
  google_api_key?: string;
};

type S3Spreadsheet = {
  sheet_source: 'S3Spreadsheet';
  file: string;
  aws_region: string;
  aws_access_key_id: string;
  aws_secret_access_key: string;
  aws_session_token?: string;
};

type SpreadsheetSource = LocalSpreadsheet | GoogleSheet | S3Spreadsheet;

export interface SheetsConfig {
  readonly spreadsheet_source: SpreadsheetSource;
  readonly sheet_page_size?: number;
  readonly stream_name?: string;
}

export class SheetsReader {
  private static sheetsReader: SheetsReader = null;
  private static sheetId: string = null; // Google Sheets ID for the current workbook

  constructor(private readonly wb: XLSX.WorkBook) {}

  static async instance(
    config: SheetsConfig,
    logger?: AirbyteLogger
  ): Promise<SheetsReader> {
    if (SheetsReader.sheetsReader) return SheetsReader.sheetsReader;

    let wb = null;

    switch (config.spreadsheet_source?.sheet_source) {
      case 'LocalSpreadsheet':
        if (!config.spreadsheet_source.file) {
          throw new VError('Please specify file path');
        }

        wb = await SheetsReader.loadFile(
          config.spreadsheet_source.file,
          logger
        );
        break;
      case 'GoogleSheet': {
        if (!config.spreadsheet_source.google_sheet_id) {
          throw new VError('Please specify Google sheet ID');
        }

        const privateKey =
          config.spreadsheet_source.google_service_account_private_key;
        const apiKey = config.spreadsheet_source.google_api_key;

        if ((privateKey && apiKey) || (!privateKey && !apiKey)) {
          throw new VError(
            'Please specify exactly one of Google account private key or Google API key'
          );
        }

        const creds = privateKey
          ? (JSON.parse(privateKey) as ServiceAccountCredentials)
          : apiKey;

        wb = await SheetsReader.loadSheets(
          config.spreadsheet_source.google_sheet_id,
          config.sheet_page_size,
          creds,
          logger
        );
        break;
      }
      case 'S3Spreadsheet': {
        if (!config.spreadsheet_source.file) {
          throw new VError('Please specify file path');
        }
        if (!config.spreadsheet_source.file.startsWith('s3://')) {
          throw new VError(
            'Please specify file path in s3://bucket/path format'
          );
        }
        if (!config.spreadsheet_source.aws_region) {
          throw new VError('Please specify AWS region');
        }
        if (!config.spreadsheet_source.aws_access_key_id) {
          throw new VError('Please specify AWS access key ID');
        }
        if (!config.spreadsheet_source.aws_secret_access_key) {
          throw new VError('Please specify AWS secret access key');
        }

        const [bucketName, ...pathParts] = config.spreadsheet_source.file
          .replace('s3://', '')
          .split('/');

        wb = await SheetsReader.loadS3(
          {
            region: config.spreadsheet_source.aws_region,
            credentials: {
              accessKeyId: config.spreadsheet_source.aws_access_key_id,
              secretAccessKey: config.spreadsheet_source.aws_secret_access_key,
              sessionToken: config.spreadsheet_source.aws_session_token,
            },
          },
          bucketName,
          pathParts.join('/'),
          logger
        );

        break;
      }
      default:
        throw new VError('Please configure a spreadsheet source');
    }

    SheetsReader.sheetsReader = new SheetsReader(wb);
    return SheetsReader.sheetsReader;
  }

  async *readRows(logger?: AirbyteLogger): AsyncGenerator<{
    sheetName: string;
    row: any;
    sheetId: string;
    id: string;
  }> {
    for (const sheetName of this.getSheetNames()) {
      logger?.info(`Reading sheet: ${sheetName}`);

      const rows = XLSX.utils.sheet_to_json(this.wb.Sheets[sheetName], {
        defval: null,
      });

      for (let i = 0; i < rows.length; i++) {
        // Generate a unique record ID for each row
        const recordId = `${SheetsReader.sheetId}_${this.normalizeSheetName(
          sheetName
        )}_${i}`;
        const row = rows[i];
        yield {sheetName, row, sheetId: SheetsReader.sheetId, id: recordId};
      }
    }
  }

  getSheetNames(): ReadonlyArray<string> {
    return this.wb.SheetNames;
  }

  normalizeSheetName(sheetName: string): string {
    return sheetName.toLowerCase().split(' ').join('_');
  }

  static async loadFile(
    file: string,
    logger?: AirbyteLogger
  ): Promise<XLSX.WorkBook> {
    logger?.info(`Opening file ${file}`);

    const wb = XLSX.readFile(file, {cellDates: true});

    logger?.info(
      `Opened file '${file}' with ${
        wb.SheetNames.length
      } sheets: ${wb.SheetNames.join(', ')}`
    );

    // get sheetId from file name (without containing folders and extension)
    SheetsReader.sheetId = this.getFileName(file);

    return wb;
  }

  private static getFileName(file: string) {
    return file.split('/').pop().split('.')[0];
  }

  static async loadSheets(
    sheetId: string,
    sheetPageSize: number,
    creds: GoogleCreds,
    logger?: AirbyteLogger
  ): Promise<XLSX.WorkBook> {
    logger?.info(`Opening Google Spreadsheet with ID ${sheetId}`);
    const doc = new GoogleSpreadsheet(sheetId);

    if (typeof creds === 'string') {
      logger?.info('Authenticating with Google using API key');
      doc.useApiKey(creds);
    } else {
      logger?.info(
        `Authenticating with Google using service account ${creds.client_email}`
      );

      logger?.info(
        `Expecting that ${creds.client_email} has Viewer access to Google Spreadsheet with ID ${sheetId}`
      );

      await doc.useServiceAccountAuth(creds);
    }

    await doc.loadInfo();
    logger?.info(
      `Opened Google Spreadsheet '${doc.title}' with ${
        doc.sheetCount
      } sheets: ${Object.keys(doc.sheetsByTitle).join(', ')}`
    );

    // Store sheets id in sheetId for returning it with rows
    SheetsReader.sheetId = sheetId;

    const wb = XLSX.utils.book_new();

    for (const sheetName of Object.keys(doc.sheetsByTitle)) {
      const sheet = await SheetsReader.loadSheet(
        doc,
        sheetName,
        sheetPageSize,
        logger
      );
      XLSX.utils.book_append_sheet(wb, sheet, sheetName);
    }

    return wb;
  }

  static async loadS3(
    s3CientConf: S3ClientConfig,
    bucket: string,
    bucketPath: string,
    logger?: AirbyteLogger
  ): Promise<XLSX.WorkBook> {
    logger?.info(`Opening file 's3://${bucket}/${bucketPath}'`);

    const s3client = new S3Client(s3CientConf);
    const res = await s3client.send(
      new GetObjectCommand({Bucket: bucket, Key: bucketPath})
    );
    const bytes = await res.Body.transformToByteArray();
    const wb = XLSX.read(bytes, {cellDates: true});

    logger?.info(
      `Opened file 's3://${bucket}/${bucketPath}' with ${
        wb.SheetNames.length
      } sheets: ${wb.SheetNames.join(', ')}`
    );

    SheetsReader.sheetId = `${bucket}/${bucketPath}`;

    return wb;
  }

  static async loadSheet(
    doc: GoogleSpreadsheet,
    sheetName: string,
    sheetPageSize: number,
    logger?: AirbyteLogger
  ): Promise<XLSX.WorkSheet> {
    const sheet = SheetsReader.getSheetOrError(doc, sheetName);

    await sheet.loadHeaderRow();
    const header = sheet.headerValues;

    const rows = await SheetsReader.getSheetRows(
      sheet,
      sheetName,
      sheetPageSize,
      logger
    );
    const aoa = [header].concat(rows.map((r) => r._rawData));
    const ws = XLSX.utils.aoa_to_sheet(aoa);

    return ws;
  }

  static getSheetOrError(
    sheet: GoogleSpreadsheet,
    name: string
  ): GoogleSpreadsheetWorksheet {
    const res = sheet.sheetsByTitle[name];
    if (!res) {
      const knownSheets = Object.keys(sheet.sheetsByTitle).join(', ');
      throw new VError(
        `Sheet named '%s' was not found. Known sheets: %s`,
        name,
        knownSheets
      );
    }
    return res;
  }

  static async getSheetRows(
    sheet: GoogleSpreadsheetWorksheet,
    name: string,
    pageSize: number,
    logger?: AirbyteLogger
  ): Promise<GoogleSpreadsheetRow[]> {
    const res: GoogleSpreadsheetRow[] = [];
    let total = 0;
    let fetchedRows = 0;
    const nonHeaderRowCount = sheet.rowCount - 1; // -1 because row count includes header row, but it's excluded from getRows
    // Fetch rows in batches of pageSize until we reach the end of the sheet
    do {
      const rows = await sheet.getRows({limit: pageSize, offset: total});
      res.push(...rows);
      fetchedRows = rows.length;
      total += fetchedRows;
    } while (fetchedRows > 0 && total < nonHeaderRowCount);

    logger?.info(`Fetched ${res.length} rows from '${name}' sheet`);
    return res;
  }
}
