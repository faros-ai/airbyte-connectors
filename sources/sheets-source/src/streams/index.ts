import {Sheets} from './sheets';

export {Sheets};
