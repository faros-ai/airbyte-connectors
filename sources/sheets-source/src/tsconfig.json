{
  "extends": "../../../.tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "outDir": "../lib",
    "experimentalDecorators": true
  }
}

