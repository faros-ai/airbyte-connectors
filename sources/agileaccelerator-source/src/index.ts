import {Command} from 'commander';
import {
  AirbyteLogger,
  AirbyteSourceBase,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {
  Agileaccelerator,
  AgileacceleratorConfig,
} from './agileaccelerator/agileaccelerator';
import {Works} from './streams';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteLogger();
  const source = new AgileacceleratorSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** AgileAccelerator source implementation. */
export class AgileacceleratorSource extends AirbyteSourceBase<AgileacceleratorConfig> {
  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(
    config: AgileacceleratorConfig
  ): Promise<[boolean, VError]> {
    try {
      const agileaccelerator = await Agileaccelerator.instance(
        config,
        this.logger
      );
      await agileaccelerator.checkConnection(config);
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: AgileacceleratorConfig): AirbyteStreamBase[] {
    return [new Works(config, this.logger)];
  }
}
