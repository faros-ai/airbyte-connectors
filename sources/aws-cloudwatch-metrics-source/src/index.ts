import {Command} from 'commander';
import {
  AirbyteSourceBase,
  AirbyteSourceLogger,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {CloudWatch, Config} from './cloudwatch';
import {Metrics} from './streams';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteSourceLogger();
  const source = new CloudWatchMetricsSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** CloudWatch Metrics source implementation. */
export class CloudWatchMetricsSource extends AirbyteSourceBase<Config> {
  get type(): string {
    return 'cloudwatch-metrics';
  }

  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(config: Config): Promise<[boolean, VError]> {
    try {
      const cloudWatch = CloudWatch.instance(config);
      await cloudWatch.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: Config): AirbyteStreamBase[] {
    return [new Metrics(config, this.logger)];
  }
}
