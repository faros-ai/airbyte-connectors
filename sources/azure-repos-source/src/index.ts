import {Command} from 'commander';
import {
  AirbyteLogger,
  AirbyteSourceBase,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {AzureRepoConfig, AzureRepos} from './azure-repos';
import {PullRequests, Repositories, Users} from './streams';
import {Commits} from './streams/commits';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteLogger();
  const source = new AzureRepoSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

/** AzureRepo source implementation. */
export class AzureRepoSource extends AirbyteSourceBase<AzureRepoConfig> {
  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }
  async checkConnection(config: AzureRepoConfig): Promise<[boolean, VError]> {
    try {
      const azureRepos = await AzureRepos.make(config, this.logger);
      await azureRepos.checkConnection();
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }
  streams(config: AzureRepoConfig): AirbyteStreamBase[] {
    return [
      new Commits(config, this.logger),
      new PullRequests(config, this.logger),
      new Repositories(config, this.logger),
      new Users(config, this.logger),
    ];
  }
}
