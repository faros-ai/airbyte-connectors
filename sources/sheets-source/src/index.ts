import {Command} from 'commander';
import {
  AirbyteSourceBase,
  AirbyteSourceLogger,
  AirbyteSourceRunner,
  AirbyteSpec,
  AirbyteStreamBase,
} from 'faros-airbyte-cdk';
import VError from 'verror';

import {SheetsConfig, SheetsReader} from './sheets-reader';
import {Sheets} from './streams';

/** The main entry point. */
export function mainCommand(): Command {
  const logger = new AirbyteSourceLogger();
  const source = new SheetsSource(logger);
  return new AirbyteSourceRunner(logger, source).mainCommand();
}

export class SheetsSource extends AirbyteSourceBase<SheetsConfig> {
  get type(): string {
    return 'sheets';
  }

  async spec(): Promise<AirbyteSpec> {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    return new AirbyteSpec(require('../resources/spec.json'));
  }

  async checkConnection(config: SheetsConfig): Promise<[boolean, VError]> {
    try {
      await SheetsReader.instance(config, this.logger);
    } catch (err: any) {
      return [false, err];
    }
    return [true, undefined];
  }

  streams(config: SheetsConfig): AirbyteStreamBase[] {
    return [new Sheets(config, this.logger)];
  }
}
