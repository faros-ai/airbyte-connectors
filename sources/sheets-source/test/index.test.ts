import {
  AirbyteLogLevel,
  AirbyteSourceLogger,
  AirbyteSpec,
  SyncMode,
} from 'faros-airbyte-cdk';
import fs from 'fs-extra';
import path from 'path';
import {VError} from 'verror';

import * as sut from '../src/index';
import {SheetsConfig, SheetsReader} from '../src/sheets-reader';

const sheets = SheetsReader.instance;

describe('index', () => {
  const logger = new AirbyteSourceLogger(
    // Shush messages in tests, unless in debug
    process.env.LOG_LEVEL === 'debug'
      ? AirbyteLogLevel.DEBUG
      : AirbyteLogLevel.INFO
  );

  beforeEach(() => {
    SheetsReader.instance = sheets;
  });

  function readResourceFile(fileName: string): any {
    return JSON.parse(fs.readFileSync(`resources/${fileName}`, 'utf8'));
  }

  test('spec', async () => {
    const source = new sut.SheetsSource(logger);
    await expect(source.spec()).resolves.toStrictEqual(
      new AirbyteSpec(readResourceFile('spec.json'))
    );
  });

  test('check connection', async () => {
    const source = new sut.SheetsSource(logger);

    async function testConnectionWithExpectedError(
      inputObject: any,
      expectedErrorMessage: string
    ): Promise<void> {
      await expect(source.checkConnection(inputObject)).resolves.toStrictEqual([
        false,
        new VError(expectedErrorMessage),
      ]);
    }

    // Test for LocalSpreadsheet case
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'LocalSpreadsheet',
        },
      },
      'Please specify file path'
    );

    // Test for GoogleSheet case with missing google_sheet_id
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'GoogleSheet',
        },
      },
      'Please specify Google sheet ID'
    );

    // Test for GoogleSheet case with both private key and API key
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'GoogleSheet',
          google_sheet_id: 'your-sheet-id',
          google_service_account_private_key: 'your-private-key',
          google_api_key: 'your-api-key',
        },
      },
      'Please specify exactly one of Google account private key or Google API key'
    );

    // Test for GoogleSheet case with neither private key nor API key
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'GoogleSheet',
          google_sheet_id: 'your-sheet-id',
        },
      },
      'Please specify exactly one of Google account private key or Google API key'
    );

    // Test for default case
    testConnectionWithExpectedError(
      {},
      'Please configure a spreadsheet source'
    );

    // Test for S3Spreadsheet case with missing file
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'S3Spreadsheet',
          aws_region: 'your-aws-region',
          aws_access_key_id: 'your-access-key-id',
          aws_secret_access_key: 'your-secret-access-key',
        },
      },
      'Please specify file path'
    );

    // Test for S3Spreadsheet case with file missing 's3://' prefix
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'S3Spreadsheet',
          file: 'your-bucket/your-path',
          aws_region: 'your-aws-region',
          aws_access_key_id: 'your-access-key-id',
          aws_secret_access_key: 'your-secret-access-key',
        },
      },
      'Please specify file path in s3://bucket/path format'
    );

    // Test for S3Spreadsheet case with missing AWS region
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'S3Spreadsheet',
          file: 's3://your-bucket/your-path',
          aws_access_key_id: 'your-access-key-id',
          aws_secret_access_key: 'your-secret-access-key',
        },
      },
      'Please specify AWS region'
    );

    // Test for S3Spreadsheet case with missing AWS access key ID
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'S3Spreadsheet',
          file: 's3://your-bucket/your-path',
          aws_region: 'your-aws-region',
          aws_secret_access_key: 'your-secret-access-key',
        },
      },
      'Please specify AWS access key ID'
    );

    // Test for S3Spreadsheet case with missing AWS secret access key
    testConnectionWithExpectedError(
      {
        spreadsheet_source: {
          sheet_source: 'S3Spreadsheet',
          file: 's3://your-bucket/your-path',
          aws_region: 'your-aws-region',
          aws_access_key_id: 'your-access-key-id',
        },
      },
      'Please specify AWS secret access key'
    );
  });

  test('reads local spreadsheet', async () => {
    const source = new sut.SheetsSource(logger);
    const streams = source.streams({
      spreadsheet_source: {
        sheet_source: 'LocalSpreadsheet',
        file: path.join(__dirname, 'test_files/test_spreadsheet.xlsx'),
      },
    } as SheetsConfig);

    const sheetsStream = streams[0];
    const sheetsIter = sheetsStream.readRecords(SyncMode.FULL_REFRESH);
    const output = [];
    for await (const rec of sheetsIter) {
      output.push(rec);
    }

    expect(output).toMatchSnapshot();
  });
});
