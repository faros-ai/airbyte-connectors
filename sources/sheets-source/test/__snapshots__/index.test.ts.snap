// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index reads local spreadsheet 1`] = `
Array [
  Object {
    "id": "test_spreadsheet_sheet1_0",
    "row": Object {
      "Age": 28,
      "Country": "USA",
      "Name": "John",
      "Occupation": "Engineer",
      "Salary": 75000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_1",
    "row": Object {
      "Age": 32,
      "Country": "Canada",
      "Name": "Emily",
      "Occupation": "Designer",
      "Salary": 62000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_2",
    "row": Object {
      "Age": 45,
      "Country": "UK",
      "Name": "Michael",
      "Occupation": "Manager",
      "Salary": 95000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_3",
    "row": Object {
      "Age": 23,
      "Country": "Germany",
      "Name": "Sara",
      "Occupation": "Student",
      "Salary": 0,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_4",
    "row": Object {
      "Age": 35,
      "Country": "France",
      "Name": "Daniel",
      "Occupation": "Writer",
      "Salary": 42000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_5",
    "row": Object {
      "Age": 29,
      "Country": "Australia",
      "Name": "Jessica",
      "Occupation": "Doctor",
      "Salary": 120000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_6",
    "row": Object {
      "Age": 31,
      "Country": "Spain",
      "Name": "Alex",
      "Occupation": "Teacher",
      "Salary": 52000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_7",
    "row": Object {
      "Age": 27,
      "Country": "Italy",
      "Name": "Olivia",
      "Occupation": "Artist",
      "Salary": 38000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_8",
    "row": Object {
      "Age": 40,
      "Country": "Japan",
      "Name": "William",
      "Occupation": "Scientist",
      "Salary": 80000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
  Object {
    "id": "test_spreadsheet_sheet1_9",
    "row": Object {
      "Age": 22,
      "Country": "Brazil",
      "Name": "Sophia",
      "Occupation": "Programmer",
      "Salary": 60000,
    },
    "sheetId": "test_spreadsheet",
    "sheetName": "Sheet1",
  },
]
`;
